wrappers/obj-c/ODWLogConfiguration.h

```diff
@@ -4,111 +4,425 @@
-@param maxTeardownUploadTimeInSec An integer that time in seconds.
+ @brief Sets max teardown upload time in seconds.
+ @param maxTeardownUploadTimeInSec An integer that time in seconds.
 */
 +(void)setMaxTeardownUploadTimeInSec:(int)maxTeardownUploadTimeInSec;
 
 /*!
-@brief Sets if trace logging to file is enabled.
-@param enableTrace True if tracing is enabled.
+ @brief Sets if trace logging to file is enabled.
+ @param enableTrace True if tracing is enabled.
 */
 +(void)setEnableTrace:(bool)enableTrace;
 
 /*!
-@brief Sets if console logging from the iOS wrapper is enabled.
-@param enableConsoleLogging True if logging is enabled.
+ @brief Sets if console logging from the iOS wrapper is enabled.
+ @param enableConsoleLogging True if logging is enabled.
 */
 +(void)setEnableConsoleLogging:(bool)enableConsoleLogging;
 
 /*!
-@brief Sets the internal SDK debugging trace level.
-@param TraceLevel one of the ACTTraceLevel values.
+ @brief Sets the internal SDK debugging trace level.
+ @param TraceLevel one of the ACTTraceLevel values.
 */
 +(void)setTraceLevel:(int)TraceLevel;
 
 /*!
-@brief Returns true if tracing is enabled.
+ @brief Returns true if tracing is enabled.
 */
 +(bool)enableTrace;
 
 /*!
-@brief Returns true if console logging is enabled.
+ @brief Returns true if console logging is enabled.
 */
 +(bool)enableConsoleLogging;
 
 /*!
-@brief Sets if inner C++ exceptions should be surfaced to Wrapper consumers.
-@param surfaceCppExceptions True if C++ exceptions should be surfaced.
+ @brief Sets if inner C++ exceptions should be surfaced to Wrapper consumers.
+ @param surfaceCppExceptions True if C++ exceptions should be surfaced.
 */
 +(void)setSurfaceCppExceptions:(bool)surfaceCppExceptions;
 
 /*!
-@brief Returns true if inner C++ exceptions are surfaced to Wrapper consumers.
+ @brief Returns true if inner C++ exceptions are surfaced to Wrapper consumers.
 */
 +(bool)surfaceCppExceptions;
 
 /*!
-@brief Sets if session timestamp should be reset after a session ends
-@param enableSessionReset True if session should be reset on session end.
+ @brief Sets if session timestamp should be reset after a session ends
+ @param enableSessionReset True if session should be reset on session end.
 */
 +(void)setEnableSessionReset:(bool)enableSessionReset;
 
 /*!
-@brief Returns true if session will be reset on session end.
+ @brief Returns true if session will be reset on session end.
 */
 +(bool)enableSessionReset;
 
 /*!
-@brief Sets the cache file path.
-@param cacheFilePath A string for the cache path.
+ @brief Sets the cache file path.
+ @param cacheFilePath A string for the cache path.
 */
 +(void)setCacheFilePath:(nonnull NSString *)cacheFilePath;
 
 /*!
-@brief Returns the cache file path.
+ @brief Returns the cache file path.
 */
 +(nullable NSString *)cacheFilePath;
 
+/*!
+ @brief Sets a config key to a string value
+ @param key A key.
+ @param value A value.
+*/
+-(void)set:(nonnull NSString *)key withValue:(nonnull NSString *)value;
```

Comment:
This method has misleading behavior. It updates a new internal instance defined at ODWLogConfiguration, while other methods like setCacheMemorySizeLimitInBytes update directly instance tracked by LogManager::GetLogConfiguration(), not the internal instance. This leads to a race condition.

Reply:
Why is this misleading? `_wrappedConfiguration` is set in private init, which is called through the static method `getLogConfigurationCopy`, which creates a *copy* of the default configuration. This instance method modified wrappedConfiguration which is a copy of the default configuration. No race condition here. Calling static methods modifies default config. Calling instance methods modified copied config. This is the same behavior as Android